extensions/applicationinsights-angular-js/src/AngularPlugin.ts

```diff
@@ -8,44 +8,39 @@ import {
 } from "@microsoft/applicationinsights-common";
 import {
     IPlugin, IConfiguration, IAppInsightsCore,
-    ITelemetryPlugin, CoreUtils, ITelemetryItem,
-    IDiagnosticLogger, _InternalMessageId, LoggingSeverity, ICustomProperties
+    ITelemetryPlugin, BaseTelemetryPlugin, CoreUtils, ITelemetryItem, ITelemetryPluginChain,
+    IProcessTelemetryContext, _InternalMessageId, LoggingSeverity, ICustomProperties
 } from "@microsoft/applicationinsights-core-js";
 import { IAngularExtensionConfig } from './Interfaces/IAngularExtensionConfig';
 
-export default class AngularPlugin implements ITelemetryPlugin {
+export default class AngularPlugin extends BaseTelemetryPlugin {
     public priority = 186;
     public identifier = 'AngularPlugin';
 
-    private _logger: IDiagnosticLogger;
     private _analyticsPlugin: IAppInsights;
-    private _nextPlugin: ITelemetryPlugin;
-    private _extensionConfig: IAngularExtensionConfig;
 
-    initialize(config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[]) {
-        this._extensionConfig =
-            config.extensionConfig && config.extensionConfig[this.identifier]
-                ? (config.extensionConfig[this.identifier] as IAngularExtensionConfig)
-                : { router: null };
-        this._logger = core.logger;
+    initialize(config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?:ITelemetryPluginChain) {
+        super.initialize(config, core, extensions, pluginChain);
+        let ctx = this._getTelCtx();
```

Comment:
Even though it's private, is there another name we can use other than "context"? This word already has meaning through the SDK. Maybe "chain" or "origin"? Not a blocker, since it can be changed in a different version